docs/03-infrastructure-deployment.md

```diff
@@ -0,0 +1,304 @@
+
+The running application:
+- **Container**: Built from `Dockerfile` in repository root
+- **Initial image**: Placeholder (updated during `azd deploy`)
+- **Resources**: 0.5 CPU, 1Gi memory
+- **Ingress**: External HTTPS with port 3000
+- **Scaling**: 1-3 replicas
+- **Registry**: Connected to ACR using managed identity
+
+#### Environment Variables:
+All configuration is injected as environment variables:
+- Azure AI Search endpoint and index name
+- Azure OpenAI instance names and model names
+- API versions for completions and embeddings
+- `SET_PASSWORDLESS=true` for managed identity authentication
+- `AZURE_CLIENT_ID` for the managed identity
+
+## Deployment Commands
+
+### Full Deployment
+
+```bash
+# Deploy everything (infrastructure + application)
+azd up
+```
+
+This runs the complete deployment pipeline:
+1. Validates configuration
+2. Provisions infrastructure (or updates if exists)
+3. Builds Docker image
+4. Pushes image to ACR
+5. Deploys to Container Apps
+6. Loads vector data
+
+### Incremental Commands
+
+```bash
+# Provision/update infrastructure only
+azd provision
+
+# Deploy application code only (infrastructure must exist)
+azd deploy
+
+# View environment variables
+azd env get-values
+
+# Set environment variable
+azd env set KEY value
+
+# Delete all resources
+azd down
+```
+
+## Bicep Infrastructure as Code
+
+The `infra/main.bicep` file uses Azure Verified Modules (AVM) for best-practice resource configuration:
+
+```bicep
+// Example: OpenAI deployment
+module openAi 'br/public:avm/res/cognitive-services/account:0.7.1' = {
+  params: {
+    name: openAiServiceName
+    kind: 'OpenAI'
+    deployments: [
+      {
+        name: gptModelName
+        model: {
+          format: 'OpenAI'
+          name: gptModelName
+          version: gptModelVersion
+        }
+      }
+    ]
+  }
+}
+```
+
+### Benefits of Bicep:
+- **Type safety**: Catch errors at compile time
+- **Intellisense**: Editor support for Azure resource properties
+- **Modularity**: Reusable modules from Azure Verified Modules
+- **Idempotent**: Safe to run multiple times
+- **Declarative**: Describe desired state, not steps
+
+## Environment Configuration
+
+### Environment Name
+
+The environment name is used to generate unique resource names:
+
+```bash
+# Resource naming pattern
+${environmentName}${resourceToken}-{service}
+
+# Example with environmentName="lang-exam"
+lang-examxxxxxxxxxxxxx-openai
+lang-examxxxxxxxxxxxxx-search
+lang-examxxxxxxxxxxxxx-rg
+```
+
+**Important**: Keep environment names short (7-10 characters) to avoid exceeding Azure resource name length limits.
+
+### Resource Token
+
+A unique 13-character hash generated from:
+- Azure subscription ID
+- Environment name
+- Azure region
+
+This ensures globally unique resource names.
+
+## Regional Deployment
+
+The application is currently configured to deploy to specific regions with OpenAI model availability:
+
+- **eastus2** (East US 2)
+- **swedencentral** (Sweden Central)
```

Comment:
Mention where to find this list in the bicep file if they want to update the regions.